plugins/modules/purefa_volume.py

```diff
@@ -834,69 +904,116 @@ def create_multi_volume(module, array, single=False):
                 add_to_pgs = []
                 for add_pg in range(0, len(module.params["add_to_pgs"])):
                     add_to_pgs.append(
-                        flasharray.FixedReference(
-                            name=module.params["add_to_pgs"][add_pg]
-                        )
+                        FixedReference(name=module.params["add_to_pgs"][add_pg])
+                    )
+                if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version):
+                    res = array.post_volumes(
+                        names=names,
+                        volume=vols,
+                        context_names=[module.params["context"]],
+                        with_default_protection=module.params[
+                            "with_default_protection"
+                        ],
+                        add_to_protection_groups=add_to_pgs,
+                    )
+                else:
+                    res = array.post_volumes(
+                        names=names,
+                        volume=vols,
+                        with_default_protection=module.params[
+                            "with_default_protection"
+                        ],
+                        add_to_protection_groups=add_to_pgs,
                     )
-                res = array.post_volumes(
-                    names=names,
-                    volume=vols,
-                    with_default_protection=module.params["with_default_protection"],
-                    add_to_protection_groups=add_to_pgs,
-                )
             else:
-                res = array.post_volumes(
-                    names=names,
-                    volume=vols,
-                    with_default_protection=module.params["with_default_protection"],
-                )
+                if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version):
+                    res = array.post_volumes(
+                        names=names,
+                        context_names=[module.params["context"]],
+                        volume=vols,
+                        with_default_protection=module.params[
+                            "with_default_protection"
+                        ],
+                    )
+                else:
+                    res = array.post_volumes(
+                        names=names,
+                        volume=vols,
+                        with_default_protection=module.params[
+                            "with_default_protection"
+                        ],
+                    )
         else:
```

Comment:
IF block is missing to handle context scenarios, should be handled.
if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version):

Reply:
Not needed as this comes from a case where DEFULT_API_VERSION is not present, which is less than CONTEXT_API_VERSION